test/unit/codegen/codegen_ispc.cpp

```diff
@@ -0,0 +1,199 @@
+                SUFFIX unit_test
+                RANGE a, b
+            }
+            ASSIGNED {
+                a
+                b
+            }
+            UNITS {
+                FARADAY = (faraday) (coulomb)
+            }
+            INITIAL {
+                a = 0.
+                b = .0
+            }
+            BREAKPOINT {
+                LOCAL x, y
+                x = 1E-18 + FARADAY * 1.2345
+                y = 1e+18 + FARADAY * .1234
+                a = x * 1.012345678901234567 + y
+                b = a + 1 + 2.0
+            }
+        )";
+
+        std::string nmodl_constants_declaration = R"(
+            /** constants used in nmodl */
+            static const uniform double FARADAY = 96485.3321233100141d;
+        )";
+
+        std::string nrn_init_state_block = R"(
+            /** initialize channel */
+            export void nrn_init_unit_test(uniform unit_test_Instance* uniform inst, uniform NrnThread* uniform nt, uniform Memb_list* uniform ml, uniform int type) {
+                uniform int nodecount = ml->nodecount;
+                uniform int pnodecount = ml->_nodecount_padded;
+                const int* uniform node_index = ml->nodeindices;
+                double* uniform data = ml->data;
+                const double* uniform voltage = nt->_actual_v;
+                Datum* uniform indexes = ml->pdata;
+                ThreadDatum* uniform thread = ml->_thread;
+
+                int uniform start = 0;
+                int uniform end = nodecount;
+                foreach (id = start ... end) {
+                    int node_id = node_index[id];
+                    double v = voltage[node_id];
+                    inst->a[id] = 0.d;
+                    inst->b[id] = 0.0d;
+                }
+            }
+
+
+            /** update state */
+            export void nrn_state_unit_test(uniform unit_test_Instance* uniform inst, uniform NrnThread* uniform nt, uniform Memb_list* uniform ml, uniform int type) {
+                uniform int nodecount = ml->nodecount;
+                uniform int pnodecount = ml->_nodecount_padded;
+                const int* uniform node_index = ml->nodeindices;
+                double* uniform data = ml->data;
+                const double* uniform voltage = nt->_actual_v;
+                Datum* uniform indexes = ml->pdata;
+                ThreadDatum* uniform thread = ml->_thread;
+
+                int uniform start = 0;
+                int uniform end = nodecount;
+                foreach (id = start ... end) {
+                    int node_id = node_index[id];
+                    double v = voltage[node_id];
+                    
+                    double x, y;
+                    x = 1d-18 + FARADAY * 1.2345d;
+                    y = 1d+18 + FARADAY * 0.1234d;
+                    inst->a[id] = x * 1.012345678901234567d + y;
+                    inst->b[id] = inst->a[id] + 1.0d + 2.0d;
+                }
+            }
+        )";
```

Comment:
Overall very nice!

I am thinking about this unit test and if it could be made smaller / simple : `float_to_string` does simple job and it would be nice if it could be tested without this whole Codegen backend. And also, whenever we change how we print initial / breakpoint block then we have to change this test.

Could we move this float_to_string to some utils or codegen_helper namespace and then just unit tests those functions?

Reply:
That's a valid point. I also though testing only the function `float_to_string` however I thought that having a test for the whole backend would also be useful and also test this function.
If we change something in the code that changes the output file then we should be careful and also alter this test if needed, right?

Reply:
2 tests? We keep this, we move float_to_string in utils and we also test that alone?

Reply:
Okay, I agree. I will take care of this. Thank you!